File cache client access silently fails if cache directories are unwritable

If the server cache is unwritable, or it's not possible to create the necessary directories, then data access from the client fails in a confusing way - the server returns no data instead of an error. For example, my cache directory is /stash, and I deliberately make it impossible to cache new files in a particular directory

```
# chmod 0111 /stash//pnfs/fnal.gov/usr/nova/data/flux/gsimple/nova_v06/nd/mn
```

then xrdcp fails like this:

```
$ XRD_LOGLEVEL=Error xrdcp -f root://fermicloud058.fnal.gov//pnfs/fnal.gov/usr/nova/data/flux/gsimple/nova_v06/nd/mn/rhc/gsimple_NOvA-ND_mn000z-200i_rp11_bs1.1_pnut_lowth_f11f093bbird_target_q34_00001.root /tmp/
[0B/330.8MB][  0%][>                                                 ][0B/s] 
[2015-07-09 16:54:54.291474 -0500][Error  ][Utility           ] The declared source size is 346873679 bytes, but received 0 bytes.

Run: [ERROR] Received corrupted data
```

and anything using the POSIX read function just gets a zero return value with no error.

If pss.trace is enabled on the server you get a message that doesn't really explain what is wrong

```
[2015-07-09 17:10:54.961823 -0500][Error  ][App               ] Prefetch::Open() can't get data-FD for /stash/pnfs/fnal.gov/usr/nova/data/flux/gsimple/nova_v06/nd/mn/rhc/gsimple_NOvA-ND_mn000z-200i_rp11_bs1.1_pnut_lowth_f11f093bbird_target_q34_00001.root /stash/pnfs/fnal.gov/usr/nova/data/flux/gsimple/nova_v06/nd/mn/rhc/gsimple_NOvA-ND_mn000z-200i_rp11_bs1.1_pnut_lowth_f11f093bbird_target_q34_00001.root
```

--- a/src/XrdFileCache/XrdFileCachePrefetch.cc
+++ b/src/XrdFileCache/XrdFileCachePrefetch.cc
@@ -236,7 +236,7 @@ bool Prefetch::Open()
       int res = m_output->Open(m_temp_filename.c_str(), O_RDWR, 0644, myEnv);
       if ( res < 0)
       {
-         clLog()->Error(XrdCl::AppMsg, "Prefetch::Open() can't get data-FD for %s %s", m_temp_filename.c_str(), lPath());
+         clLog()->Error(XrdCl::AppMsg, "Prefetch::Open() can't open local file %s", m_temp_filename.c_str());
          delete m_output;
          m_output = NULL;
          return false;
@@ -951,7 +951,7 @@ Prefetch::Read(char *buff, off_t off, size_t size)
       XrdSysCondVarHelper monitor(m_stateCond);
 
       // AMT check if this can be done once during initalization
-      if (m_failed) return 0;
+      if (m_failed) return m_input.Read(buff, off, size);
 
       if ( ! m_started)
       {

